main.js

```diff
@@ -0,0 +1,283 @@
+class WeatherApp {
+  constructor() {
+    this.APIKEY = '16806dd9a591b25a5beebeb69dd718b8';
+    this.coordinates = {};
+    this.weatherInfo = {};
+    this._init();
+  }
+
+  _init() {
+    const searchForm = document.querySelector('.weather__search-form');
+    searchForm.addEventListener('submit', () => this._checkLocation(event));
+  }
+
+  async _checkLocation(event) {
+    event.preventDefault();
+    const locationInput = document.querySelector('.weather__textbox').value;
+
+    if (!locationInput) {
+      return;
+    }
+
+    await this._getCoordinates(locationInput);
+  }
+
+  async _getCoordinates(location) {
+    try {
+      this._startLoading();
+      const coordResponse = await fetch(`http://api.openweathermap.org/geo/1.0/direct?q=${location}&limit=10&appid=${this.APIKEY}`);
+
+      // TODO:400系ごとのハンドリングを分ける
+      if (!coordResponse.ok) {
+        switch (coordResponse.status) {
+          case 401:
+            throw new Error400('401 error');
+          case 404:
+            throw new Error400('404 error');
+          case 429:
+            throw new Error400('429 error');
+          case 500, 502, 503, 504:
+            throw new Error('500~504 error');
+          default:
+            throw new Error('somthing went wrong');
+        }
+      }
+      const coordArr = await coordResponse.json();
+
+      if (coordArr.length === 0) {
+        const noCityHtml = 'お探しの都市の天気情報が見つかりませんでした。<br>別の都市名を入力し、再度お試しください。'
+        this._displayError(noCityHtml);
+        return;
+      }
+
+      if (coordArr.length >= 2) {
+        this._getMultiLocationNames(coordArr);
+        return;
+      }
+
+      this.coordinates = {
+        lat: coordArr[0].lat,
+        lon: coordArr[0].lon,
+      }
+
+      await this._getWeather(this.coordinates);
+
+    } catch (e) {
+      console.error(`${e}`);
+
+      if (e instanceof Error400) {
+        const errorHtml = '天気情報が取得できませんでした。';
+        this._displayError(errorHtml);
+        return;
+
+      } else {
+        const errorHtml = '天気情報が取得できませんでした。<br>時間をおいてから再度お試しください。';
+        this._displayError(errorHtml);
+        return;
+      }
+
+    } finally {
+      this._endLoading();
+    }
+
+  }
+
+  // APIの仕様で都市名が完全一致しない場合は複数の候補を返してくることがある
+  // 複数の候補が返って来た場合はユーザーに選択させる方法がベストと考えた
+  _getMultiLocationNames(coordArr) {
+    const multiLoc = [];
+    coordArr.forEach(location => {
+      if (location.local_names && location.local_names.ja) {
```

Comment:
[Q]
この条件分岐はなぜ必要なのですか？？

Reply:
@cyndiii29 
なるべく都市名を日本語名で表示をしたいからです。

APIで返却される都市の情報にlocal_names（各国の言語の都市名）が含まれるかは都市次第、更にその中にlocal_names.ja（日本語名）が含まれるかも都市次第なので、この分岐を追加しています。
どちらも存在しない場合は、必ず含まれるlocation.nameを採用するようにしています。

この背景がコードだけだと分かりづらいと思うのでコメントで追加しておきます。

併せてリファクタリングができるか検討します。

Reply:
@cyndiii29 
・分岐を入れて理由をコメントで追加
・重複しているコードを別メソッドに切り出しました

Reply:
良いコメントですね！

またその意図を汲み取るなら
```suggestion
      if (location.local_names.ja) {
        // 日本語都市名
      }
      else if (location.location_names) {
          // 日本語じゃない都市名
      }
      else {
          // location
      }
```

これのほうがよくないですか？？
@KenJamesHioki

Reply:
@cyndiii29 
コード内の僕のコメントがミスリードする内容だったので、まずは訂正させてください。

//都市の日本語名が表示可能な場合は日本語名の表示を優先するための分岐

と書いていますが、正しくは、

//都市の日本語名が表示可能な場合は日本語名の表示を優先し、なければ必ず英語名をするための分岐

更に追加情報：
locationオブジェクトにはlocal_namesがある場合とない場合がある。
local_namesが存在する場合でも、ja（日本語名）がある場合とない場合がある。また英語名もある場合とない場合がある。

そのため、local_namesがある&&ja（日本語名）がある場合は、その日本語名を取得。
ない場合は英語名が担保されているnameを取得するように実装をしています。

Reply:
[IMO]
なるほどですね！
その旨はコードにコメントしておかないと、他のエンジニアも同じ勘違いをする可能性があるので書いてある方が個人的には良いのかなと思いました！